### ISSUE TITLE
boolean type triggers error in BayesianNetwork.from_samples when using pandas dataframe

### ISSUE THREAD

**Issue by handsaway** [jmschrei/pomegranate#362]
I have a pandas dataframe containing booleans.
I'm converting the dataframe to an array to use it as training data for a Bayesian network.

When I convert the dataframe to an array **without touching the types**, then BayesianNetwork.from_samples() gives an error.
There is no more error if I convert the whole data frame **enforcing a string type for all variables**.

See the minimal working example below:

```
import numpy as np
import pandas as pd
from pomegranate import *

data1 = [[True, 'A', 'A', 'C', 1, True],
        [True, 'A', 'A', 'C', 0, True],
        [False, 'A', 'A', 'B', 1, False],
        [False, 'A', 'A', 'A', 2, False],
        [False, 'A', 'A', 'C', 1, False],
        [False, 'B', 'B', 'B', 2, False],
        [False, 'B', 'B', 'C', 0, False],
        [True, 'C', 'C', 'A', 2, True],
        [True, 'C', 'C', 'C', 1, False],
        [True, 'C', 'C', 'C', 0, False],
        [True, 'C', 'C', 'C', 2, True],
        [True, 'C', 'B', 'A', 1, False]]

df = pd.DataFrame(data1)
data2 = np.array(df)
data3 = np.array(df, dtype=np.dtype('U25'))

model = BayesianNetwork.from_samples(data1, name='testBool', \
                                     state_names=['a', 'b', 'c', 'd', 'e', 'f'])
```
Using `data2` or `df` as argument triggers the error:
```
Traceback (most recent call last):
  File "/Users/xxxx/Documents/PycharmProjects/tests/learningWithBoolean.py", line 25, in <module>
    state_names=['a', 'b', 'c', 'd', 'e', 'f'])
  File "pomegranate/BayesianNetwork.pyx", line 954, in pomegranate.BayesianNetwork.BayesianNetwork.from_samples
KeyError: 'True'
```
While using `data1` (which contains booleans) or `data3` (which doesn't contain booleans, only strings) works correctly. I'd like to be able to keep booleans for integrity, if possible. Am I missing something?

PS: I used the data array from the huge monty hall example (pomegranate/examples/bayesnet_huge_monty_hall.ipynb) for this example.

Edit: Added `df` triggering the error also.

**Comment by jmschrei** [comment]
No, this looks like an issue on my side. Thanks for pointing it out. However, it is kind of low priority for now given the other things I am trying to add in. Is this a blocking issue for you?

**Comment by handsaway** [comment]
Not blocking but I have to add a logic in lots of places to switch between strings and booleans. So a fix will be appreciated! Thanks for the heads up @jmschrei

**Comment by jmschrei** [comment]
This should be fixed in 0.12.0 and the use of data generators.
